processor/src/lib.rs

```diff
@@ -488,31 +505,99 @@ where
         self.chiplets.get_mem_value(ctx, addr)
     }
 
-    pub fn into_parts(self) -> (System, Decoder, Stack, RangeChecker, Chiplets, A) {
+    pub fn into_parts(self) -> (System, Decoder, Stack, RangeChecker, Chiplets, H) {
         (
             self.system,
             self.decoder,
             self.stack,
             self.range,
             self.chiplets,
-            self.advice_provider,
+            self.host.into_inner(),
         )
     }
 }
 
+// PROCESS STATE
+// ================================================================================================
+
+/// A trait that defines a set of methods which allow access to the state of the process.
+pub trait ProcessState {
+    /// Returns the current clock cycle of a process.
+    fn clk(&self) -> u32;
+
+    /// Returns the current execution context ID.
+    fn ctx(&self) -> u32;
+
+    /// Returns the value located at the specified position on the stack at the current clock cycle.
+    fn get_stack_item(&self, pos: usize) -> Felt;
+
+    /// Returns a word located at the specified word index on the stack.
+    ///
+    /// Specifically, word 0 is defined by the first 4 elements of the stack, word 1 is defined
+    /// by the next 4 elements etc. Since the top of the stack contains 4 word, the highest valid
+    /// word index is 3.
+    ///
+    /// The words are created in reverse order. For example, for word 0 the top element of the
+    /// stack will be at the last position in the word.
+    ///
+    /// Creating a word does not change the state of the stack.
+    fn get_stack_word(&self, word_idx: usize) -> Word;
+
+    /// Returns stack state at the specified clock cycle. This includes the top 16 items of the
+    /// stack + overflow entries.
+    ///
+    /// # Panics
+    /// Panics if invoked for non-last clock cycle on a stack instantiated with
+    /// `keep_overflow_trace` set to false.
```

Comment:
I don't think that is the true.  The current implementation uses the [`get_state_at`](https://github.com/0xPolygonMiden/miden-vm/blob/main/processor/src/stack/mod.rs#L124-L135) method from the `Stack` object.

Reply:
A more recent implementation of this method is [here](https://github.com/0xPolygonMiden/miden-vm/blob/next/processor/src/stack/mod.rs#L130). If invoked for the current clock cycle it will always succeed, but if invoked for some other clock cycle it will fail unless the stack was instantiated with `keep_overflow_trace=true`, which happens only when we create the process in debug mode.

Looking at this more, I'm wondering if we should change `ProcessState::get_stack_state_at()` to something like:
```Rust
fn get_stack_state(&self) -> Vec<Felt> {
    self.stack.get_state_at(self.system.clk())
}
```